### ISSUE TITLE
Training overfitting on custom dataset

### ISSUE THREAD

**Issue by wwidjaya** [deepinsight/insightface#1395]
Dear @nttstar and colleagues, 

I have prepared a custom face dataset similar to LFW with 600 identities and 12K images for my research, and able to generate the dataset based on the predefined format.

I am able to run a training session based on the dataset with defaults network: RsNet100. the training accuracy quickly climbing to 100%, and the verification accuracy increases up to 0.15688 (15.6%) in epoch 12

The following are the snapshot of the training session result:

```
call reset()
2021-01-25 13:38:28,608 root        : INFO     Epoch[12] Batch [0-20]	Speed: 67.70 samples/sec	acc=1.000000	lossvalue=0.000000
2021-01-25 13:38:38,010 root        : INFO     Epoch[12] Batch [20-40]	Speed: 68.10 samples/sec	acc=1.000000	lossvalue=0.000000
2021-01-25 13:38:47,437 root        : INFO     Epoch[12] Batch [40-60]	Speed: 67.91 samples/sec	acc=1.000000	lossvalue=0.000000
lr-batch-epoch: 0.1 75 12
testing verification..
(5606, 512)
infer time 51.81577400000001
[ilfw][4000]XNorm: 18.111734
[ilfw][4000]Accuracy-Flip: 0.11371+-0.10493
testing verification..
(5322, 512)
infer time 48.87921500000002
[ilfw-test][4000]XNorm: 18.014994
[ilfw-test][4000]Accuracy-Flip: 0.15668+-0.06784
saving 1
2021-01-25 13:40:40,766 root        : INFO     Saved checkpoint to "./models/r100-arcface-emore/model-0001.params"
[4000] Accuracy-Highest: 0.15668
2021-01-25 13:40:42,840 root        : INFO     Epoch[12] Batch [60-80]	Speed: 5.55 samples/sec	acc=1.000000	lossvalue=0.000000
2021-01-25 13:40:52,221 root        : INFO     Epoch[12] Batch [80-100]	Speed: 68.24 samples/sec	acc=1.000000	lossvalue=0.000000
2021-01-25 13:41:01,634 root        : INFO     Epoch[12] Batch [100-120]	Speed: 68.01 samples/sec	acc=1.000000	lossvalue=0.000000
```

The verification accuracy never increased ever since.

The dataset is being prepared by the following process:

a. Collect data from google images, resized to 112x112, and aligned
b. Remove duplicates
c. Add an augmented face masks
d. Split dataset of persons into training, validation, and test (80:10:10). Persons in training sets never used in validation, vice versa.
e. Generate a .lst file based on the training dataset
f. Based on the generated .lst file, create .rec and .idx file
g. Generate matcher and unmatched pairs file based on validation and test sets
h. generate bin file based on pairs file

My questions are:
1. Am I making any mistakes in data preparation steps?
2. How many identities are required as a minimum to achieve good results?

Thank you for your kind help and support.

Best Regards,

Wirianto Widjaya

**Comment by wwidjaya** [comment]
So sorry, I made a terrible mistake on the data label. will close the issue instead. Thanks

**Comment by ThanhPham1987** [comment]
Hi @wwidjaya,
Would you like to share how to prepare dataset for training?
Best regards,
PeterPham
